.github/workflows/generate-html-for-pull-request.yaml

```diff
@@ -16,33 +16,53 @@ jobs:
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-      
+
+      - name: Check if enough lines were changed
+        id: enough_lines
+        run: |
+          git checkout ${{ github.head_ref }}
+
+          LATEST=$(git rev-parse HEAD)
+          SECOND_LATEST=$(git rev-parse HEAD~1)
+
+          LINES_CHANGED=$(git diff --shortstat $SECOND_LATEST $LATEST | awk '{print $4 + $6}')
+
+          echo "Lines changed: $LINES_CHANGED"
+
+          if [ "$LINES_CHANGED" -gt 20 ]; then
+            echo "continue=true" >> $GITHUB_ENV
+          else
+            echo "continue=false" >> $GITHUB_ENV
+          fi
+
       - name: Find directories with changed rmd or image files
+        if: env.continue == 'true'
         id: changed-files-dir-names
         uses: tj-actions/changed-files@v41
         with:
           files:  |
             src/G*/*.{rmd,Rmd}
             src/G*/img/*.{png,gif}
-            src/G*/img/*/*.{png,gif}            
+            src/G*/img/*/*.{png,gif}
```

Comment:
Do we have situations where someone puts a folder into the img-folder and inside that folder the pictures? Or is there some other reason for this?

Reply:
Almost all courses have their exercise- specific images in their own folder (harjoitus_1 etc.), which is why we need to check for subfolders as well. Also when using CourseCrafter it specifically sorts every image to a folder according to the exercise.